spec/integration/outputs/delete_spec.rb

```diff
@@ -0,0 +1,61 @@
+require_relative "../../../spec/es_spec_helper"
+require "logstash/outputs/elasticsearch"
+
+
+describe "Versioned delete", :integration => true, :version_greater_than_equal_to_2x => true do
+  require "logstash/outputs/elasticsearch"
+
+  def get_es_output( options={} )
+    settings = {
+      "manage_template" => true,
+      "index" => "logstash-delete",
+      "template_overwrite" => true,
+      "hosts" => get_host_port(),
+    }
+    LogStash::Outputs::ElasticSearch.new(settings.merge!(options))
+  end
+
+  before :each do
+    @es = get_client
+    # Delete all templates first.
+    # Clean ES of data before we start.
+    @es.indices.delete_template(:name => "*")
+    # This can fail if there are no indexes, ignore failure.
+    @es.indices.delete(:index => "*") rescue nil
+    @es.indices.refresh
+  end
+
+  context "when delete only" do
+    it "should ignore non-monotonic external_version updates" do
+      id = "ev2"
+      subject = get_es_output({ 'document_id' => id, "external_version" => "99", "action" => "index" })
+      subject.register
+      subject.multi_receive([LogStash::Event.new("message" => "foo")])
+      r = @es.get(:index => 'logstash-delete', :type => 'logs', :id => id, :refresh => true)
+      insist { r["_version"] } == 99
```

Comment:
allrighty, on the dynamic version bit... It looks like the variable substitution isn't happening.

I tried testing with this snippet in delete_spec.rb:

```
  context "when delete only" do
    it "should ignore non-monotonic external version updates" do
      id = "ev2"
      subject = get_es_output({ 'document_id' => id, "version" => "%{my_version}", "version_type" => "external", "action" => "index" })
      subject.register
      subject.multi_receive([LogStash::Event.new("message" => "foo", "my_version" => 99)])
      ...
```

I get this exception: 

```
Failures:

  1) Versioned delete when delete only should ignore non-monotonic external version updates
     Failure/Error: LogStash::Outputs::ElasticSearch.new(settings.merge!(options))

     LogStash::ConfigurationError:
       Something is wrong with your configuration.
     # ./vendor/bundle/jruby/1.9/gems/logstash-core-5.1.2.snapshot1-java/lib/logstash/config/mixin.rb:125:in `config_init'
     # ./vendor/bundle/jruby/1.9/gems/logstash-core-5.1.2.snapshot1-java/lib/logstash/outputs/base.rb:63:in `initialize'
     # ./spec/integration/outputs/delete_spec.rb:15:in `get_es_output'
     # ./spec/integration/outputs/delete_spec.rb:31:in `(root)'
     # ./vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
```

So I put a "puts" in mixin.rb to see what's up:

```
    if !self.class.validate(params)
      puts "params[#{params}]"
      raise LogStash::ConfigurationError,
        I18n.t("logstash.runner.configuration.invalid_plugin_settings")
    end
```

And this is what it spits out:

```
Versioned delete
  when delete only
params[{"manage_template"=>true, "index"=>"logstash-delete", "template_overwrite"=>true, "hosts"=>[//127.0.0.1:9200], "document_id"=>"ev2", "version"=>"%{my_version}", "version_type"=>"external", "action"=>"index", "id"=>"elasticsearch_647b6be3-dddd-4bce-8464-9ee0d84f4463", "enable_metric"=>true, "type"=>"", "tags"=>[], "exclude_tags"=>[], "codec"=><LogStash::Codecs::Plain id=>"plain_9bc0c20a-233c-4b26-b5b2-b669d1177838", enable_metric=>true, charset=>"UTF-8">, "workers"=>1, "template_name"=>"logstash", "parent"=>nil, "idle_flush_time"=>1, "upsert"=>"", "doc_as_upsert"=>false, "script"=>"", "script_type"=>"inline", "script_lang"=>"painless", "script_var_name"=>"event", "scripted_upsert"=>false, "retry_initial_interval"=>2, "retry_max_interval"=>64, "retry_on_conflict"=>1, "pipeline"=>nil, "ssl_certificate_verification"=>true, "sniffing"=>false, "sniffing_delay"=>5, "timeout"=>60, "failure_type_logging_whitelist"=>[], "pool_max"=>1000, "pool_max_per_route"=>100, "healthcheck_path"=>"/", "resurrect_delay"=>5, "validate_after_inactivity"=>10000}]
    should ignore non-monotonic external version updates (FAILED - 1)
    should commit monotonic external version updates

Failures:

  1) Versioned delete when delete only should ignore non-monotonic external version updates
     Failure/Error: LogStash::Outputs::ElasticSearch.new(settings.merge!(options))

     LogStash::ConfigurationError:
       Something is wrong with your configuration.
     # ./vendor/bundle/jruby/1.9/gems/logstash-core-5.1.2.snapshot1-java/lib/logstash/config/mixin.rb:126:in `config_init'
     # ./vendor/bundle/jruby/1.9/gems/logstash-core-5.1.2.snapshot1-java/lib/logstash/outputs/base.rb:63:in `initialize'
     # ./spec/integration/outputs/delete_spec.rb:15:in `get_es_output'
     # ./spec/integration/outputs/delete_spec.rb:31:in `(root)'
     # ./vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
```

So you can see it's trying to initalize the "version" property to "%{my_version}", which fails validation as a number.

What am I doing wrong here?

Thanks.